src/v/redpanda/admin_server.cc

```diff
@@ -349,6 +349,45 @@ void admin_server::log_request(
       req.get_url());
 }
 
+void admin_server::log_exception(
+  const ss::sstring& url,
+  const request_auth_result& auth_state,
+  std::exception_ptr eptr) const {
+    using http_status = ss::httpd::reply::status_type;
+    using http_status_ut = std::underlying_type_t<http_status>;
+    const auto log_ex = [&](int status) {
+        std::stringstream os;
+        const auto username
+          = (auth_state.get_username().size() > 0 ? auth_state.get_username() : "_anonymous");
+        if (status > 0) {
+            fmt::print(os, "[{}] {} - {}", username, url, status, eptr);
+        } else {
+            fmt::print(os, "[{}] - {}", username, url, eptr);
+        }
+        return os.str();
+    };
+
+    try {
+        std::rethrow_exception(eptr);
+    } catch (ss::httpd::base_exception& ex) {
+        const auto status = static_cast<http_status_ut>(ex.status());
+        if (status >= 400 && status <= 499) {
+            vlog(logger.warn, "{}", log_ex(status));
+        } else if (status >= 500) {
+            static constexpr auto exemptions = std::to_array<http_status>(
+              {http_status::service_unavailable});
```

Comment:
I think we can simplify this by flipping the logic around, and say that _only_ 500 is an error, and all other http return codes are warnings, rather than saying that >500 codes are errors unless they're in an allow list.

Reason: the principle at work is that ERROR is to be used for times _we've_ done something wrong (which is the meaning of 500, the server goofed), and WARN is for everything else weird/unexpected happening.  (I'm biased cos I wrote it, but: https://vectorizedio.atlassian.net/wiki/spaces/CORE/pages/146243598/Logging+from+Redpanda+Core)